Sources/Purchasing/Offering.swift

```diff
@@ -314,16 +314,22 @@ extension Offering {
     /// - Throws: Error if the content couldn't be deserialized to the expected type.
     /// - Note: This decodes JSON using `JSONDecoder.KeyDecodingStrategy.convertFromSnakeCase`.
     public func getMetadataValue<T: Decodable>(for key: String) -> T? {
-        do {
-            guard let value = self.metadata[key] else { return nil }
-            let data = try JSONSerialization.data(withJSONObject: value)
-
-            return try JSONDecoder.default.decode(
-                T.self,
-                jsonData: data,
-                logErrors: true
-            )
-        } catch {
+        guard let value = self.metadata[key] else { return nil }
+
+        if JSONSerialization.isValidJSONObject(value) {
+            do {
+                let data = try JSONSerialization.data(withJSONObject: value)
+                return try JSONDecoder.default.decode(
+                                T.self,
+                                jsonData: data,
+                                logErrors: true
+                            )
+            } catch {
+                return nil
+            }
```

Comment:
I think we can simplify this a bit with `try?` instead of `try`, right?

Reply:
totally, I tried to retain as much existing as possible but happy to simplify